mpmp/src/model/Player.java

```diff
@@ -83,6 +85,26 @@ public boolean addMoney(int sum) {
 	}
 
 	/**
+	 * Collect a sum of money from each player.
+	 */
+	public void collect(int sum) {
```

Comment:
This is the part where it gets ugly. We don't have a loan system (we need one), so this code and `addMoney` are inherently broken anyway. To city my copy of the rules (1992 German edition):

> 
> ##### Bankrott
> 
> Ein Spieler, der die Zahlungen einstellt, d.h. einer, der mehr schuldet als er bezahlen kann, muß dem Gläubiger alles, was er an Wert hat, übergeben und darf nicht mehr weiterspielen. Häuser und Hotels müssen auch in diesem Fall zum halben Preis an die Bank verkauft werden, die das Geld an den Gläubiger zahlt. Wenn ein zahlungsunfähiger Spieler ein hypothekarisch belastetes Besitztum übergibt, muß der neue Besitzer sofort 10 Prozent Zinsen an die Bank zahlen; er kann gleichzeitig die Hypothek zurückzahlen.
> 
> Wenn ein Spieler nicht genügend Geld aufbringt, um seine Steuern und Strafen zu zahlen, selbst wenn er seine Häuser verkauft und sein Besitztum hypothekarisch belastet, übernimmt der Bank sein Hab und Gut und versteigert alles an den Meistbietenden. Häuser und Hotels dürfen niht mitversteigert werden.